<?xml version="1.0" ?>
<launch>
  
  <arg name="emulate_laser" default="true"/>
  <arg name="bag" default="false"/>
  <arg name="tf" default="true"/>
  <arg name="infrared" default="true"/>
  <arg name="rgbd_odom" default="false"/>
  <arg name="gas_sensor" default="true"/>
<!--   <arg name="data_skip" default="0"/> -->

  
  <!-- Camera stuff -->
  <include file="$(find openni2_multicamera)/launch/7_cam/7_cam.launch" >
<!--     <arg name="data_skip" value="$(arg data_skip)"/>     -->
  </include>
  
  <include file="$(find seekthermal_ros)/launch/seekthermal_ros.launch" if="$(arg infrared)"/>
  <node name="flipper" type="flip_image_node" pkg="flip_image" output="screen" if="$(arg infrared)" respawn="true">
    <remap from="/input_image" to="/thermal_camera/image_colored"/>
  </node>
  
  <!-- TF stuff (in the future will be inside the driver. TODO: add parameter -->
  <include file="$(find siar_driver)/launch/frames_7_cam_new_platform.launch" if="$(arg tf)"/>
  
  <include file="$(find siar_driver)/launch/bag.launch" if="$(arg bag)"/>
  
  <node name="gas_sensor" type="libelium_waspmote_node.py" pkg="libelium_waspmote_gas_node" if="$(arg gas_sensor)" />
  
  <node name="rgbd_odom_node" type="rgbd_odom_node" pkg="rgbd_odom" if="$(arg rgbd_odom)">
    <param name="camera_topic" value="/front" />
    <param name="tf_topic_out" value="/rgbd_odom/transform" />
    <param name="max_features" value="800" />
    <param name="flow_threshold" value="10" />
    <param name="min_matches" value="15" />
    <param name="odom_frame_id" value="odom" />
    <param name="base_frame_id" value="base_link" />
    <param name="publish_image" value="true" />
    <param name="publish_point_cloud" value="true" />
    <param name="use_sba" value="false" />
    <param name="type_sba" value="1" />         # 0: MOTIONSTRUCTURE, 1: MOTION, 2: STRUCTURE (cvsba)
    <param name="max_keyframes" value="10" />
    <param name="use_imu" value="true" />
    <param name="imu_topic" value="/arduimu_v3/imu" />
    <param name="apply_downsampling" value="false" />
    <param name="pc_downsampling" value="0.1" />
    <param name="apply_crop" value="true" />
    <param name="pc_crop_dist" value="7.0" />
    <param name="apply_outlier" value="false" />
    <param name="pc_outlier_radius" value="0.4" />
    <param name="pc_outlier_neighbors" value="10" />
    <param name="ground_height_topic" value="/ground_estimator_node/ground_height" />
    <param name="ground_alpha" value="0.8" />
  </node>
  
</launch>
